votingsystem/src/voting.rs

```diff
@@ -0,0 +1,270 @@
+    authorization_token: u64,
+    used_in: VoteId,
+    voter_publickey: Public,
+}
+
+impl VotePaper {
+    pub fn new(id: GeneralMeetingId, agenda_number: u32, token: u64, pub_key: Public) -> Self {
+        let mut rng = rand::thread_rng();
+        let random_id = rng.gen();
+        let vote_id = VoteId {
+            id: random_id,
+        };
+        Self {
+            general_meeting_id: id,
+            agenda_number,
+            authorization_token: token,
+            used_in: vote_id,
+            voter_publickey: pub_key,
+        }
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+struct Vote {
+    vote_paper_id: BallotId,
+    choice: VoteResult,
+    signature: Signature,
+}
+
+impl Vote {
+    pub fn new(choice: VoteResult, signature: Signature) -> Self {
+        let mut rng = rand::thread_rng();
+        let random_id = rng.gen();
+        let ballot_id = BallotId {
+            id: random_id,
+        };
+        Self {
+            vote_paper_id: ballot_id,
+            choice,
+            signature,
+        }
+    }
+}
+struct Context {
+    pub storage: Option<Box<dyn SubStorageAccess>>,
+}
+
+impl Context {
+    fn storage(&self) -> &dyn SubStorageAccess {
+        self.storage.as_ref().unwrap().as_ref()
+    }
+
+    fn storage_mut(&mut self) -> &mut dyn SubStorageAccess {
+        self.storage.as_mut().unwrap().as_mut()
+    }
+}
+
+impl Service for Context {}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct TxCreateVotePaper {
+    general_meeting_id: GeneralMeetingId,
+    agenda_number: u32,
+    authorization_token: u64,
+    voter_public_key: Public,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Ballot {
+    choice: VoteResult,
+    voter_signature: Signature,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum TxAction {
+    CreateVotePaper(TxCreateVotePaper),
+    Vote(Ballot),
+}
+
+impl Action for TxAction {}
+pub type OwnTransaction = crate::common::SignedTransaction<TxAction>;
+
+impl Context {
+    fn excute_tx(&mut self, transaction: &Transaction) -> Result<Id, ExecuteError> {
+        if transaction.tx_type() != "Vote" {
+            return Err(ExecuteError::InvalidMetadata)
+        }
+        let tx: OwnTransaction =
+            serde_cbor::from_slice(&transaction.body()).map_err(|_| ExecuteError::InvalidFormat)?;
+        tx.verify().map_err(|_| ExecuteError::InvalidSign)?;
+
+        let mut meeting_id = &GeneralMeetingId {
+            id: 0,
+        };
+        match tx.tx.action {
+            TxAction::CreateVotePaper(TxCreateVotePaper {
+                general_meeting_id,
+                agenda_number,
+                authorization_token,
+                voter_public_key,
+            }) => {
+                let vote_paper =
+                    VotePaper::new(general_meeting_id.clone(), agenda_number, authorization_token, voter_public_key);
+                self.storage_mut().set(vote_paper.used_in.as_ref(), serde_cbor::to_vec(&vote_paper).unwrap());
```

Comment:
When we create a vote paper, the vote_paper_used_in will be empty because it is not used yet.